server/uploader.js

```diff
@@ -108,20 +115,38 @@ uploader.uploadFile = function(desc, file, prefix, bigWigToWigAddr,
  * @param {string} prefix Folder that contains the bw file.
  * @param {string} bwFile Name of the bigwig file (without prefix).
  * @param {string} bigWigToWigAddr The convention script path.
- * @param {string} uploadPath Directory of temporary files.
+ * @param {!mongodb.Db} db The database object.
+ * @param {function(*)} callback The callback function.
  */
-uploader.bigWigToBCWig = function(prefix, bwFile, bigWigToWigAddr, uploadPath) {
+uploader.bigWigToBCWig = function(prefix, bwFile, bigWigToWigAddr, db,
+                                  callback) {
   // convert *.bw into *.wig
-  var storedName = bwFile + '.data';
+  var percentage = 0;
+
   var wigFileName = bwFile + '.wig';
   log.serverLog('start transfer');
   var cmd = [
     bigWigToWigAddr,
-    prefix + storedName,
+    prefix + bwFile,
     prefix + wigFileName
   ].join(' ');
   childProcess.execSync(cmd);
   log.serverLog(cmd);
+  percentage += 10;
+  dbData.updateProgress(db, bwFile, percentage, function(err) {
+    if (err.error) {
+      callback(err);
+    }
+  });
+
+  cmd = [
+    'wc',
+    '-l',
+    prefix + wigFileName
+  ].join(' ');
+  var output = childProcess.execSync(cmd).toString().split(RegExp(/\s+/));
```

Comment:
childProcess.exec() could by an async version, which is potentially better? You may mark a TODO here.